File: src/TypeFormatter/VoidTypeFormatter.ts
import type { Definition } from "../Schema/Definition.js";
import type { SubTypeFormatter } from "../SubTypeFormatter.js";
import type { BaseType } from "../Type/BaseType.js";
import { VoidType } from "../Type/VoidType.js";

export class VoidTypeFormatter implements SubTypeFormatter {
    public supportsType(type: BaseType): boolean {
        return type instanceof VoidType;
    }
    public getDefinition(type: VoidType): Definition {
        return { type: "null" };
    }
    public getChildren(type: VoidType): BaseType[] {
        return [];
    }
}

File: internal/formatter/void_type_formatter.go
package formatter

import (
	"github.com/vega/ts-json-schema-generator/internal/schema"
	"github.com/vega/ts-json-schema-generator/internal/types"
)

// VoidTypeFormatter mirrors src/TypeFormatter/VoidTypeFormatter.ts.
type VoidTypeFormatter struct{}

func NewVoidTypeFormatter() *VoidTypeFormatter { return &VoidTypeFormatter{} }

func (f *VoidTypeFormatter) SupportsType(t types.Type) bool {
	_, ok := t.(*types.VoidType)
	return ok
}

func (f *VoidTypeFormatter) GetDefinition(t types.Type) *schema.Definition {
	return &schema.Definition{Type: "null"}
}

func (f *VoidTypeFormatter) GetChildren(t types.Type) []types.Type { return nil }
